tests/test_pass_fds.py

```diff
@@ -0,0 +1,96 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2018
+# Author: Xu Han <xuhan@redhat.com>
+
+import time
+import unittest
+
+from aexpect import client
+
+
+SLEEP = 1
+
+DEVNULL = "/dev/null"
+LIST_FD_CMD = ('''python -c "import os; os.system('ls -l /proc/%d/fd' % '''
+               '''os.getpid())"''')
+
+
+class PassfdsTest(unittest.TestCase):
+
+    def test_pass_fds_spawn(self):
+        """
+        Tests fd passing for `client.Spawn`
+        """
+        with open(DEVNULL, "r") as devnull:
+            fd = devnull.fileno()
+
+            child = client.Spawn(LIST_FD_CMD)
+            time.sleep(SLEEP)
```

Comment:
Hmm, I really hate sleeps. How about `child.get_status()` instead? That will makes certain the command finished (and you can also check the status to avoid false-positives).

Reply:
ok, makes sense, let me check it.